create vm from snapshot example.py

```diff
@@ -0,0 +1,263 @@
+    vm_config['name'] = f'{connection_name}'
+    vm_config['machineType'] = \
+        f'projects/{project_id}/zones/{server_location}/machineTypes/{machine_type}'
+    vm_config['zone'] = f'projects/{project_id}/zones/{server_location}'
+    vm_config['metadata']['items'][0]['value'] = vm_startup_script
+    vm_config['disks'][0]['deviceName'] = f'{connection_name}'
+    vm_config['disks'][0]['diskType'] = \
+        f'projects/{project_id}/zones/{server_location}/diskTypes/pd-ssd'
+    vm_config['networkInterfaces'][0]['subnetwork'] = \
+        f'projects/{project_id}/regions/europe-west1/subnetworks/default'
+    # NOTE: two sources can be used for initial system installation:
+    # a) from 'sourceSnapshot'
+    # b) from 'sourceImage'
+    # In the example .json the parameter 'sourceSnapshot' is defined. 
+    # This flow script shall install Ubuntu which is provided from a 'sourceImage'.
+    # Since only either of the two options can be given as a parameter, 
+    # the paramter 'sourceSnapshot' must be removed 
+    # (either here, see below, or in the .json).
+    find_image_name = this.flow('find build-server image name', wait=False)
+    image_name = find_image_name.wait().get('output_value')['image_name']
+    vm_config['disks'][0]['initializeParams'][
+        'sourceImage'] = f'projects/ubuntu-os-cloud/global/images/{image_name}'
+    try:
+        del vm_config['disks'][0]['initializeParams']['sourceSnapshot']
+    except KeyError:
+        pass
+        
+    # Connect to the VM
+    operation = this.connect(
+        'gcloud',
+        name=f'launch {connection_name}',
+        api_name='compute',
+        api_version='v1',
+        collection='instances',
+        request='insert',
+        params={
+            'project': project_id,
+            'zone': server_location,
+            'body': vm_config,
+        },
+    ).get('output_value')['result']
+    this.flow(
+        'google_operation_wait',
+        operation=operation,
+    )
+
+    try:
+        # retrieve external IP and hostkey (required for creating a SSH connection):
+        for _ in range(60):
+            instance_list = this.connect(
+                'gcloud',
+                name=f'find {connection_name}',
+                api_name='compute',
+                api_version='v1',
+                collection='instances',
+                request='list',
+                params={
+                    'project': project_id,
+                    'zone': server_location,
+                    'filter': f'name={connection_name}',
+                },
+            ).get('output_value')
+            if len(instance_list['result'].get('items', [])) != 0:
+                break
+            this.sleep(1)
+        else:
+            return this.error('did not find VM within 60 tries')
+        external_ip = instance_list['result']['items'][0]['networkInterfaces'][0]['accessConfigs'][0]['natIP']
+
+        failed_tries = []
+        for _ in range(60):
```

Comment:
please use this schema for all retry loops:
```python
last_try = None
for _ in range(20):
    if last_try is not None:
        last_try.archive()
    try:
        last_try = this.task(
            ...
        )
    except flow_api.DependencyFailedError:
        this.sleep(30)
    else:
        break
else:
    return this.error('some-error-message')
```
```suggestion
        for _ in range(60):
```